packages/client/src/Components/MessageCreator/MessageCreator.tsx

```diff
@@ -82,7 +85,129 @@ const MessageCreator: React.FC<Props> = (props) => {
     }
   }, [props])
 
+  /**
+   * helper function to render default Datetime or Date or Time props of json
+   */
+  const configCommonProps = (prop: any) => {
+    switch (prop.format) {
+      case "datetime-local":
+        prop.default = moment(gameDate).format("DD/MM/YYYY HH:mm")
+        prop.options !== undefined ? prop.options.flatpickr = flatpickr(".calendar") : prop.options = {}
+        prop.options = {"flatpickr": {
+          "wrap":false,
+          "time_24hr": true,
+          "dateFormat":"d/m/Y H:i",
+        }}
+        return prop
+      case "date":
+        prop.default = moment(gameDate).format("DD/MM/YYYY")
+        prop.options !== undefined ? prop.options.flatpickr = flatpickr(".calendar") : prop.options = {}
+        
+        prop.options = {"flatpickr": {
+          "wrap":false,
+          "dateFormat":"d/m/Y",
+        }}
+        return prop
+      case "time":
+        prop.default = moment(gameDate).format("HH:mm")
+        prop.options !== undefined ? prop.options.flatpickr = flatpickr(".calendar") : prop.options = {}
+        
+        prop.options = {"flatpickr": {
+          "wrap":false,
+          "time_24hr": true,
+          "dateFormat":"H:i",
+        }}
+        return prop
+      default:
+        return prop
+    }
+  }
+
+  /**
+   * Render Default datetime entries in template of json for type datetime-local
+   */
+  const configDateTimeLocal = (schema: any) => {
+    if(!schema || !schema.properties){
+      return
+    }
+    Object.keys(schema.properties).forEach(key => {
+      let prop = schema.properties[key]
+      if(prop.format === 'datetime-local' || prop.format === 'date' || prop.format === 'time'){
+        prop = configCommonProps(prop)
+      }
+      
+      if(prop.type === 'object'){
+        configDateTimeLocal(prop)
+      }else if(prop.type === 'array'){
+        configDateTimeLocal(prop.items)
```

Comment:
in the above line, we pass in an array of items.  Will this still work with the `configDateTimeLocal` method?  (it looks like that only processes one line at a time)

Reply:
https://github.com/serge-web/serge/blob/develop/packages/client/src/Schemas/DailyIntentions.json#L3
https://github.com/serge-web/serge/blob/develop/packages/client/src/Schemas/DailyIntentions.json#L15

Daily Intension schema template has type array and object as "properties" which will work in both cases.

Reply:
Yes, but don't we need to provide iterator for `array`?

Reply:
![image](https://user-images.githubusercontent.com/16476120/126986653-422b19ac-6ae6-4a00-8007-7165f794af22.png)
![image](https://user-images.githubusercontent.com/16476120/126986733-3299a170-a4a2-44f7-b698-e32ca59d0bb0.png)

Json Schema Editor specification mentioned that {"type":"array"} will take array in "items.properties" which is ultimately an object which is mentioned in image 1.

Reply:
Aah, I see.  That's fine then :-D